### PR TITLE
Highlight change ids in rebase advice

### PR DESCRIPTION
**Pull request by Veykril**
Fixes https://github.com/martinvonz/jj/issues/3561

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
  - I don't know if that needs mention as a new feature or not, let me know if it does
- [ ] ~~I have updated the documentation (README.md, docs/, demos/)~~
- [ ] ~~I have updated the config schema (cli/src/config-schema.json)~~
- [x] I have ~~added~~updated tests to cover my changes

### CODE REVIEW

**Review by martinvonz — APPROVED** [review]
Thanks! This should make my demo at Git Merge next week a tiny bit prettier :)

File: cli/src/cli_util.rs [hunk 1]
```diff
@@ -1290,6 +1290,11 @@ impl WorkspaceCommandHelper {
             .expect("parse error should be confined by WorkspaceCommandHelper::new()")
     }
 
+    pub fn format_short_change_id(&self) -> TemplateRenderer<'_, Commit> {
```
Comment (yuja) [inline cli/src/cli_util.rs:1293]: nit: `format_()` sounds like a function that returns formatted object. maybe rename to `short_change_id_template()`?

File: cli/src/cli_util.rs [hunk 2]
```diff
@@ -1290,6 +1290,11 @@ impl WorkspaceCommandHelper {
...
+        self.parse_commit_template("format_short_change_id(self.change_id())")
+            .expect("syntax is valid")
```
Comment (yuja) [inline cli/src/cli_util.rs:1295]: `format_short_change_id()` can be overridden by user, and may have syntax error.

Comment (Veykril) [reply cli/src/cli_util.rs:1295]: Ah okay, I thought it only parsed this snippet specifically (anything else being caught upfront)

**Comment by martinvonz** [comment]
> I don't know if that needs mention as a new feature or not, let me know if it does

Nah, I don't think so. I think in general, it makes sense to mention things that fit at least one of:

* Would break or surprise existing users' workflows
* Users would appreciate but would perhaps not notice if we don't mention it
* Are significant enough that users who don't yet use jj are encouraged to finally try it (or is enough to convince them to try it again)

There may be other good reasons, and that's just my opinion anyway.

**Pull request merged**